### PR TITLE
Disable timeline events collection in Ray by default

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Profiling adds memory and CPU overhead, and is a central bottleneck in the GCS, so disable it by default. This significantly reduces the memory overhead in certain cases. For example, a worker using ray.get in a loop goes from 2GB->80MB heap usage after disabling profiling.

This only affects ray.timeline().

### CODE REVIEW

**Comment by edoakes** [comment]
@ericl post microbenchmark results?

File: doc/source/troubleshooting.rst [hunk 1]
```diff
@@ -14,6 +14,9 @@ The most important tool is the timeline visualization tool. To visualize tasks
 in the Ray timeline, you can dump the timeline as a JSON file by running ``ray
 timeline`` from the command line or by using the following command.
 
+To use the timeline, Ray profiling must be enabled by setting the
+``RAY_PROFILING=1`` environment variable prior to starting Ray.
```
Comment (edoakes) [inline doc/source/troubleshooting.rst:18]:
```suggestion
To use the timeline, Ray profiling must be enabled by setting the
``RAY_PROFILING=1`` environment variable prior to starting Ray on every machine.
```

**Review by edoakes — APPROVED** [review]
Looks good, have you measured overhead/perf gain?

**Comment by ericl** [comment]
```
(profiling off)

single client get calls per second 32853.31 +- 53.43 (14% faster)
single client put calls per second 22654.6 +- 127.72 (15% faster)
multi client put calls per second 112783.22 +- 1748.81 (20% faster)
single client get calls (Plasma Store) per second 7876.59 +- 64.29 (5% faster)
single client put calls (Plasma Store) per second 4610.29 +- 14.58 (4% faster)
multi client put calls (Plasma Store) per second 7092.47 +- 212.08 (~)
single client tasks sync per second 1008.68 +- 9.12 (~)
single client tasks async per second 9239.47 +- 64.45 (~)
multi client tasks async per second 24928.78 +- 133.16 (~)
1:1 actor calls sync per second 1441.01 +- 9.35 (9% faster)
1:1 actor calls async per second 5860.68 +- 94.18 (~)
1:1 actor calls concurrent per second 5372.6 +- 57.66 (~)
1:n actor calls async per second 12197.64 +- 280.68 (~)
n:n actor calls async per second 30860.4 +- 913.01 (10% faster)
n:n actor calls with arg async per second 9375.51 +- 132.17 (8% faster)
1:1 async-actor calls sync per second 1051.55 +- 26.19 (~)
1:1 async-actor calls async per second 3121.94 +- 11.63 (4% faster)
1:1 async-actor calls with args async per second 1961.29 +- 14.52 (~)
1:n async-actor calls async per second 10597.17 +- 139.72 (~)
n:n async-actor calls async per second 22058.11 +- 524.95 (8% faster)

(profiling on)

single client get calls per second 28894.86 +- 165.94
single client put calls per second 19677.01 +- 201.43
multi client put calls per second 92498.19 +- 2406.35
single client get calls (Plasma Store) per second 7430.8 +- 90.72
single client put calls (Plasma Store) per second 4404.31 +- 74.46
multi client put calls (Plasma Store) per second 7046.89 +- 176.88
single client tasks sync per second 983.54 +- 7.4
single client tasks async per second 9332.83 +- 156.82
multi client tasks async per second 24805.92 +- 389.2
1:1 actor calls sync per second 1320.28 +- 10.52
1:1 actor calls async per second 5752.98 +- 374.73
1:1 actor calls concurrent per second 5325.42 +- 96.98
1:n actor calls async per second 12648.1 +- 105.38
n:n actor calls async per second 27979.3 +- 1168.38
n:n actor calls with arg async per second 8634.83 +- 67.07
1:1 async-actor calls sync per second 1178.43 +- 19.63
1:1 async-actor calls async per second 2994.88 +- 53.98
1:1 async-actor calls with args async per second 1968.7 +- 18.87
1:n async-actor calls async per second 10293.93 +- 71.44
n:n async-actor calls async per second 20353.99 +- 913.89
```

Looks like 5-10% speedup on bottlenecked workloads.

**Comment by richardliaw** [comment]
Can we also please remove ray.timeline from our tutorials, etc then?

https://github.com/search?l=Python&q=ray.timeline&type=Code

A quick search shows a couple uses of the API call, largely in Ray/Anyscale sponsored material.

**Comment by ericl** [comment]
Need to fix tests

File: python/ray/state.py [hunk 2]
```diff
@@ -519,6 +519,11 @@ def chrome_tracing_dump(self, filename=None):
 
                 all_events.append(new_event)
 
+        if not all_events:
+            logger.warning(
+                "No profiling events found. Ray profiling must be enabled "
+                "by setting RAY_PROFILING=1.")
+
```
Comment (ijrsvt) [inline python/ray/state.py:526]: @ericl Can you verify that this is propagated when Ray Client is used?

**Pull request merged**